#! /bin/sh -e
## 10_sysctl_quiet by Simon Horman
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Adds quiet -q flag to sysctl

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch}"

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
       -patch) patch $patch_opts -p1 < $0;;
       -unpatch) patch $patch_opts -p1 -R < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argum
ent"
                exit 1;;
esac

exit 0
@DPATCH@
diff -ru procps-3.1.8/sysctl.8 procps-3.1.8.new/sysctl.8
--- procps-3.1.8/sysctl.8	2002-12-12 07:51:32.000000000 +0900
+++ procps-3.1.8.new/sysctl.8	2003-04-15 20:30:46.000000000 +0900
@@ -12,9 +12,9 @@
 .SH SYNOPSIS
 .B "sysctl [-n] [-e] variable ..."
 .br
-.B "sysctl [-n] [-e] -w variable=value ..."
+.B "sysctl [-n] [-e] [-q] -w variable=value ..."
 .br
-.B "sysctl [-n] [-e] -p <filename>"
+.B "sysctl [-n] [-e] [-q] -p <filename>"
 .br
 .B "sysctl [-n] [-e] -a"
 .br
@@ -45,6 +45,9 @@
 .B "-e"
 Use this option to ignore errors about unknown keys.
 .TP
+.B "-q"
+Use this option to not display the values set to stdout.
+.TP
 .B "-w"
 Use this option when you want to change a sysctl setting.
 .TP
diff -ru procps-3.1.8/sysctl.c procps-3.1.8.new/sysctl.c
--- procps-3.1.8/sysctl.c	2002-12-29 08:26:50.000000000 +0900
+++ procps-3.1.8.new/sysctl.c	2003-04-15 20:35:08.000000000 +0900
@@ -15,6 +15,8 @@
  * Changelog:
  *            v1.01:
  *                   - added -p <preload> to preload values from a file
+ *            Horms: 
+ *                   - added -q to be quiet when modifying values
  *
  * Changes by Albert Cahalan, 2002.
  */
@@ -46,6 +48,7 @@
 static bool PrintName;
 static bool PrintNewline;
 static bool IgnoreError;
+static bool Quiet;
 
 /* error messages */
 static const char ERR_UNKNOWN_PARAMETER[] = "error: Unknown parameter '%s'\n";
@@ -80,9 +83,9 @@
  */
 static int Usage(const char *restrict const name) {
    printf("usage:  %s [-n] [-e] variable ... \n"
-          "        %s [-n] [-e] -w variable=value ... \n" 
+          "        %s [-n] [-e] [-q] -w variable=value ... \n" 
           "        %s [-n] [-e] -a \n" 
-          "        %s [-n] [-e] -p <file>   (default /etc/sysctl.conf) \n"
+          "        %s [-n] [-e] [-q] -p <file>   (default /etc/sysctl.conf) \n"
           "        %s [-n] [-e] -A\n", name, name, name, name, name);
    return -1;
 }
@@ -290,13 +293,14 @@
       fprintf(fp, "%s\n", value);
       fclose(fp);
 
-      if (PrintName) {
+      if(Quiet) {
+	 ;
+      } else if (PrintName) {
          fprintf(stdout, "%s = %s\n", outname, value);
+      } else if (PrintNewline) {
+         fprintf(stdout, "%s\n", value);
       } else {
-         if (PrintNewline)
-            fprintf(stdout, "%s\n", value);
-         else
-            fprintf(stdout, "%s", value);
+         fprintf(stdout, "%s", value);
       }
    }
 
@@ -377,6 +381,7 @@
    PrintName = true;
    PrintNewline = true;
    IgnoreError = false;
+   Quiet = false;
 
    if (argc < 2) {
        return Usage(me);
@@ -419,6 +424,9 @@
                  preloadfile = *argv;
               }
               return Preload(preloadfile);
+	 case 'q':
+	      Quiet = true;
+	   break;
          case 'a': /* string and integer values (for Linux, all of them) */
          case 'A': /* the above, including "opaques" (would be unprintable) */
          case 'X': /* the above, with opaques completly printed in hex */
